core/src/main/java/org/apache/iceberg/encryption/Ciphers.java

```diff
@@ -101,37 +117,52 @@ public AesGcmDecryptor(byte[] keyBytes) {
     }
 
     public byte[] decrypt(byte[] ciphertext, byte[] aad) {
-      int plainTextLength = ciphertext.length - GCM_TAG_LENGTH - NONCE_LENGTH;
+      return decrypt(ciphertext, 0, ciphertext.length, aad);
+    }
+
+    public byte[] decrypt(
+        byte[] ciphertext, int ciphertextOffset, int ciphertextLength, byte[] aad) {
       Preconditions.checkState(
-          plainTextLength >= 1,
+          ciphertextLength - GCM_TAG_LENGTH - NONCE_LENGTH >= 1,
```

Comment:
Should there be a similar check in the encrypt method to validate that the plaintext has at least one byte?